### DISCUSSION TITLE
Supabase:Algolia Sync

### DISCUSSION THREAD

**Discussion by dommarr** [supabase/supabase#638] [Questions] [answered]
Hi, 

When a change occurs in my Supabase table, I'd like to trigger a function to update an Algolia index. In the Vercel documentation it mentions using Supabase for realtime data events ([link](https://vercel.com/docs/solutions/realtime)).

I see the subscribe option in Supabase, but not sure that is meant for this use case. Any thoughts on the best way to handle this?

Alternatively I can setup a cron job to periodically sync, but I'd prefer the above solution if possible.

Thank you!

**Comment by inian** [comment, marked as answer]
We will launch Supabase functions during the last week of March which is perfect for this use case, so stay tuned!

In the meantime, you could write a database function using `pg_cron` and `psql_http` which is installed in your Supabase DB to trigger the sync. I am assuming there is a way to update the Algolia index via a REST API call. Or a typical cron job as you mentioned would work too. 

If you want the index to be updated in realtime, you can use our client libraries to subscribe to changes and send it to Algolia as they arrive. This would require the script to be running continuously on a server somewhere (until we launch functions).

**Reply by dommarr** [reply]
Thanks @inian! Looking forward to Supabase functions. If you need an early adopter/tester, let me know. 

For anyone curious, I wrote a Vercel serverless function and I will use a cron job to hit the endpoint a few times a day. Will probably use Github Actions (free for my use case), but haven't set it up yet. [Link to Vercel docs](https://vercel.com/docs/solutions/cron-jobs).

**Reply by inian** [reply]
Amazing, thanks for the offer @dommarr! We will get in touch soon.
